lib/support/browserScript.js

```diff
@@ -46,6 +47,54 @@ async function getScriptsForCategories(categories) {
   return result;
 }
 
+/**
+ * Take the name of a script, its location on the disk
+ * and its contents and generate a ScriptObject:
+ *
+ * name: The name of the script.
+ * function: A function object if this is a new-style
+ *           script; null otherwise.
+ * content: The source code of the function (contents)
+ *          if this is a old-style script; null otherwise.
+ * requires: An object with boolean fields indictating
+ *           which requirements must be met to execute
+ *           this new-style function; the object is
+ *           empty for an old-style function.
+ *
+ * @param {string}
+ * @param {string}
+ * @param {string}
+ * @returns {ScriptObject}
+ */
+function generateScriptObject(name, path, contents) {
+  try {
+    const scriptAndMetadataObject = require(path);
+
+    if (
+      typeof scriptAndMetadataObject.function === 'function' &&
+      typeof scriptAndMetadataObject.requires === 'object'
+    ) {
+      log.info(name + ' is an new-style script object.');
+      return {
+        name: name,
+        requires: scriptAndMetadataObject.requires,
+        function: scriptAndMetadataObject.function,
+        content: null
+      };
+    }
+  } catch (error) {
+    // Use this as a signal, but don't perform any action
```

Comment:
Can you say how the signal is used?  You fall back to an old style script?

Reply:
Great suggestion!